Formula/gitversion.rb

```diff
@@ -0,0 +1,26 @@
+class Gitversion < Formula
+  desc "Easy semantic versioning for projects using Git"
+  homepage "https://github.com/GitTools/GitVersion"
+  url "https://www.nuget.org/api/v2/package/GitVersion.CommandLine/3.6.2"
+  version "3.6.2"
+  sha256 "db05ebacfb76ddca26180f0c70060fbb89800e434d3d55b524785d0da37dbac8"
+
+  bottle :unneeded
+
+  depends_on "mono"
+
+  def install
+    file = File.new("gitversion", "w", 0555)
+    file.write("#!/bin/sh\n")
+    file.write("#{HOMEBREW_PREFIX}/opt/mono/bin/mono #{libexec}/tools/GitVersion.exe \"$@\"\n")
+    file.flush
+    file.close
+
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/gitversion"
+  end
+
+  test do
+    system "#{bin}/gitversion", "-?"
```

Comment:
`test do` is already in a temporary repository so you could do all the rest of these with `system` commands inside the `test do` block. There's other `git*` formulae that do similar things in their `test do` block.